lib/pluginmanager/remove.rb

```diff
@@ -25,12 +25,7 @@ class LogStash::PluginManager::Remove < LogStash::PluginManager::Command
   def execute
     signal_error("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting") unless File.writable?(LogStash::Environment::GEMFILE_PATH)
 
-    ##
-    # Need to setup the bundler status to enable uninstall of plugins
-    # installed as local_gems, otherwise gem:specification is not
-    # finding the plugins
-    ##
-    LogStash::Bundler.setup!({:without => [:build, :development]})
+    LogStash::Bundler.prepare({:without => [:build, :development]})
```

Comment:
(_this probably needs to be rolled back, committed to open a discussion_)
 
In `LogStash::Bundler.setup!` we are executing two processes: prepare and `::Bundle:setup`. It seems to me `Bundle:setup` using system bundler. `./gradlew clean assembleOssZipDistribution` produces `Operation aborted, cannot remove plugin, message: can't find gem bundler (= 2.3.26) with executable bundle` error
As per [bundler doc](https://bundler.io/guides/bundler_setup.html#setting-up-your-application-to-use-bundler), setup is required to load the `Gemfile` dependencies which seems loading dependencies well as I debugged.

Issue entry point: `Bundler#setup` -> `Runtime#setup` -> [`SharedHelpers#set_bundle_environment`](https://github.com/rubygems/rubygems/blob/v3.3.26/bundler/lib/bundler/shared_helpers.rb#L276-L279) tries to find either `../../exe/bundle` or `../../libexec/bundle` and fails. (Rubygem version 3.3.26: https://github.com/rubygems/rubygems/blob/v3.3.26/bundler)

However, running process with rake (`rake artifact:achives_oss`) is fine and rake process seems using system `../../libexec/bundle` which passing the error lines.

We are not using gradle tasks in pipelines but we mention in the [_development_ docs](https://github.com/mashhurs/logstash/blob/96a9e3b3149fbcdea18969c9337371f88a0b8b5c/README.md?plain=1#L214), which needs to be fixed near future.

Reply:
I think it is related to this open issue: https://github.com/elastic/logstash/issues/15284